src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -483,6 +507,28 @@ function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
     return constraints_MP
 end
 
+# preconditions have been checked in the caller function
+function _linear_map_hrep_lift(M::AbstractMatrix{N}, P::AbstractPolyhedron{N}) where {N<:Real}
+    m, n = size(M)
+
+    # we extend M to an invertible m x m matrix by appending m-n columns
+    # orthogonal to the column space of M
+    Mext, inv_Mext = extend(M, check_rank=false)
+
+    # append zeros to the existing constraints, in the last m-n coordinates
+    cext = [HalfSpace(vcat(c.a, zeros(N, m-n)), c.b) for c in constraints_list(P)]
```

Comment:
I wanted to suggest to outsource `vcat(c.a, zeros(N, m-n))` to a function call `extend(c.a, m-n)` and define that function elsewhere. But maybe that is overkill. As you prefer.

Reply:
I think it is readable with a `vcat` (but there's always a personal bias on such things, of course). While we are at it, it is true that if we preallocate the `cext` and write a for loop this has some chance of giving some "free" efficiency.